### DISCUSSION TITLE
Remaining question in second session about GIT branches

### DISCUSSION THREAD

**Discussion by bogdanbilcan** [koushikkothagal/java-backend-bootcamp#21] [Sessions Q&A]
If you branch the main repo for a new feature and in between the team is working and finishes a different feature or fixes a bug that affects your code too, how can we sync our branch with main which now is ahead of our feature branch.
How to treat the merge problems, that might occur if the bug or new feature implemented before mine, is actually affecting the same classes, pieces of code as the one i am already working on.
How can we avoid code regression or messing it up entirely ? 
For example messing up our branch and forcing us to re-branch and start from scratch the coding ?

**Comment by somixyz** [comment]
If i understand correctly, you should pull changes from that branch. 
So when you are in your working branch you should pull from main. I assume that your upstream branch is main, and your branch is lets say here abc-branch then you should do `git pull origin main`  (origin is name of repo, it could be set different name and if it is then you should that one) after that as you mentioned you could have CONFLICT with your state and state u just received from main. 
You should resolved conflict issue and and finish merge process.
